server/validate/email/validate.go

```diff
@@ -79,30 +165,68 @@ func (v *validator) Init(jsonconf string) error {
 		v.auth = smtp.PlainAuth("", v.senderEmail, v.SenderPassword, v.SMTPAddr)
 	}
 
-	// If a relative path is provided, try to resolve it relative to the exec file location,
-	// not whatever directory the user is in.
-	if !filepath.IsAbs(v.ValidationTemplFile) {
-		basepath, err := os.Executable()
-		if err == nil {
-			v.ValidationTemplFile = filepath.Join(filepath.Dir(basepath), v.ValidationTemplFile)
-		}
-	}
-	if !filepath.IsAbs(v.ResetTemplFile) {
-		basepath, err := os.Executable()
-		if err == nil {
-			v.ResetTemplFile = filepath.Join(filepath.Dir(basepath), v.ResetTemplFile)
-		}
-	}
+	// Optionally resolve paths against the location of this executable file.
+	v.ValidationTemplFile = resolveTemplatePath(v.ValidationTemplFile)
+	v.ResetTemplFile = resolveTemplatePath(v.ValidationTemplFile)
 
-	v.htmlValidationTempl, err = ht.ParseFiles(v.ValidationTemplFile)
+	// Paths to templates could be templates themselves: they may be language-dependent.
+	var validationPathTempl, resetPathTempl *textt.Template
+	validationPathTempl, err = textt.New("validation").Parse(v.ValidationTemplFile)
 	if err != nil {
 		return err
 	}
-	v.htmlResetTempl, err = ht.ParseFiles(v.ResetTemplFile)
+	resetPathTempl, err = textt.New("reset").Parse(v.ResetTemplFile)
 	if err != nil {
 		return err
 	}
 
+	var path string
+	if len(v.Languages) > 0 {
+		v.validationTempl = make(map[int]*textt.Template, len(v.Languages))
+		v.resetTempl = make(map[int]*textt.Template, len(v.Languages))
+
+		var langTags []i18n.Tag
+		// Find actual content templates for each defined language.
+		for idx, lang := range v.Languages {
+			tag, err := i18n.Parse(lang)
+			if err != nil {
+				return err
+			}
+			langTags = append(langTags, tag)
+			if v.validationTempl[idx], path, err = readTemplateFile(validationPathTempl, lang); err != nil {
+				return err
+			}
+			if err = isTemplateValid(v.validationTempl[idx]); err != nil {
+				return fmt.Errorf("parsing %s: %w", path, err)
+			}
+
+			if v.resetTempl[idx], path, err = readTemplateFile(resetPathTempl, lang); err != nil {
+				return err
+			}
+			if err = isTemplateValid(v.resetTempl[idx]); err != nil {
+				return fmt.Errorf("parsing %s: %w", path, err)
+			}
+		}
+		v.langMatcher = i18n.NewMatcher(langTags)
+	} else {
+		// No i18n support. Use defaults.
+		v.validationTempl[0], path, err = readTemplateFile(validationPathTempl, "")
+		if err != nil {
+			return err
+		}
+		if err = isTemplateValid(v.validationTempl[0]); err != nil {
+			return fmt.Errorf("parsing %s: %w", path, err)
+		}
+
+		v.resetTempl[0], path, err = readTemplateFile(resetPathTempl, "")
+		if err != nil {
+			return err
+		}
+		if err = isTemplateValid(v.resetTempl[0]); err != nil {
+			return fmt.Errorf("parsing %s: %w", path, err)
```

Comment:
Ditto.

Reply:
Done

Reply:
Wrong earlier comment. Leaving as is because the error is more generic than invalid path.